### ISSUE TITLE
Using IfcOpenShell with VisualStudio and C++ language

### ISSUE THREAD

**Issue by pierrenavarra** [IfcOpenShell/IfcOpenShell#1379]
Hi,
I'd like to use IfcOpenShell with VisualStudio with C++ language because I'd like to use it in Unreal Engine.
At the moment, I wrote my code with Python. It works. 
Into unreal environnement, I can run my Python script and passing arguments, etc.
The problem is : in Unreal, no way to get Python Script into BluePrint.
But in Unreal, it is possible to code in C++ and call function into BluePrint. 
How to use ifcOpenShell in C++?
Can't find it.
Thx

**Comment by aothms** [comment]
Have a look at https://github.com/IfcOpenShell/IfcOpenShell#using-visual-studio and https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.6.0/src/examples/IfcOpenHouse.cpp https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.6.0/src/examples/IfcParseExamples.cpp

**Comment by pierrenavarra** [comment]
Thx aothms for this exemples
The think is not clear for me is the process to install ifcopenShell into Visual Studio.
Could you help me a little bit?
In visual Studio using C# it's easy to add GitUb package but with C++ I don't know why... Sorry maybe it's a little bit ridiculous...

**Comment by aothms** [comment]
This is all a matter of personal preference. You could use CMake, to setup your project in a cross-platform fashion. Or create a visual studio solution and manually set the header and library paths and files. A visual studio C++ tutorial on the web should cover this. It's not a matter of installation, but setting the correct paths in the MSVC project properties.

**Comment by pierrenavarra** [comment]
> Or create a visual studio solution and manually set the header and library paths and files. 
Do you mean, for exemple, if I just want to open and ifc file a get the version, I'll have to copy the src files like wich one? 
>but setting the correct paths in the MSVC project properties. 
What do you mean ?
Sorry to keep your time..

**Comment by aothms** [comment]
Not not source, IfcOpenShell src is going to be compiled into static library files (.lib on windows). These provide the implementation. Then the header files the declarations (API definition if you wish). Setting the correct paths will make sure that MSVC can find these files.

**Comment by pierrenavarra** [comment]
Hi.
I've installed Git : version 2.30.2windows.1
Also CMake 3.20.0-rc4
I've downloaded the last sourcesCodes 
When I'm with powerShell (as admin) into win direxctory I've got this error....

`

bluids-deps.cmd: The term "bluids-deps.cmd" is not recognized as the name of cmdlet, function, file
script or executable program. Check the spelling of the name, or if a path exists, verify that the path
access is correct and try again.
At character Line: 1: 1
+ bluids-deps.cmd
+ ~~~~~~~~~~~~~~~
    + CategoryInfo: ObjectNotFound: (bluids-deps.cmd: String) [], CommandNotFoundException
    + FullyQualifiedErrorId: CommandNotFoundException`
